containers/unit_tests/TestDualView.hpp

```diff
@@ -463,12 +472,36 @@ TEST(TEST_CATEGORY, dualview_deep_copy) {
   test_dualview_deep_copy<double, TEST_EXECSPACE>();
 }
 
+struct NoDefaultConstructor {
+  NoDefaultConstructor(int i_) : i(i_) {}
+
+  operator int() const { return i; }
+  NoDefaultConstructor& operator+=(const NoDefaultConstructor& other) {
+    i += other.i;
+    return *this;
+  }
+
+  int i;
+};
+}  // namespace Test
+
+template <>
+struct Kokkos::reduction_identity<Test::NoDefaultConstructor> {
+  static Test::NoDefaultConstructor sum() { return {0}; }
+};
+
+namespace Test {
+
 TEST(TEST_CATEGORY, dualview_realloc) {
   test_dualview_realloc<int, TEST_EXECSPACE>();
+  Impl::test_dualview_realloc<NoDefaultConstructor, TEST_EXECSPACE,
+                              /* Initialize */ false>();
```

Comment:
A little bit disturbing to have `Impl::` on this line and not on the one above.

Reply:
`test_dualview_realloc` calls `Impl::test_dualview_realloc` both with `Initialiize=true` and `Initialize=false`. We can only test with `Initialize=false` for `NoDefualtConstructor`, though.